lib/ansible/module_utils/docker_common.py

```diff
@@ -0,0 +1,498 @@
+#
+#  Copyright 2016 Red Hat | Ansible
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import logging
+import re
+import json
+import sys
+import copy
+
+from requests.exceptions import SSLError
+from urlparse import urlparse
+from logging import Handler, NOTSET
+from ansible.module_utils.basic import *
+
+HAS_DOCKER_PY = True
+HAS_DOCKER_ERROR = None
+
+try:
+    from docker import Client
+    from docker import __version__ as docker_version
+    from docker.errors import APIError, TLSParameterError, NotFound
+    from docker.tls import TLSConfig
+    from docker.constants import DEFAULT_TIMEOUT_SECONDS, DEFAULT_DOCKER_API_VERSION
+    from docker.utils.types import Ulimit, LogConfig
+except ImportError, exc:
+    HAS_DOCKER_ERROR = str(exc)
+    HAS_DOCKER_PY = False
+
+DEFAULT_DOCKER_HOST = 'unix://var/run/docker.sock'
+DEFAULT_TLS = False
+DEFAULT_TLS_VERIFY = False
+MIN_DOCKER_VERSION = "1.7.0"
+
+DOCKER_COMMON_ARGS = dict(
+    docker_host=dict(type='str', aliases=['docker_url']),
+    tls_hostname=dict(type='str'),
+    api_version=dict(type='str', aliases=['docker_api_version']),
+    timeout=dict(type='int'),
+    cacert_path=dict(type='str', aliases=['tls_ca_cert']),
+    cert_path=dict(type='str', aliases=['tls_client_cert']),
+    key_path=dict(type='str', aliases=['tls_client_key']),
+    ssl_version=dict(type='str'),
+    tls=dict(type='bool'),
+    tls_verify=dict(type='bool'),
+    debug=dict(type='bool', default=False),
+    filter_logger=dict(type='bool', default=False),
+    log_path=dict(type='str', default='docker.log'),
+    log_mode=dict(type='str', choices=['stderr', 'file', 'syslog'], default='syslog'),
+)
+
+DOCKER_MUTUALLY_EXCLUSIVE = [
+    ['tls', 'tls_verify']
+]
+
+DOCKER_REQUIRED_TOGETHER = [
+    ['cert_path', 'key_path']
+]
+
+DEFAULT_DOCKER_REGISTRY = 'https://index.docker.io/v1/'
+EMAIL_REGEX = '[^@]+@[^@]+\.[^@]+'
+BYTE_SUFFIXES = ['B', 'KB', 'MB', 'GB', 'TB', 'PB']
+
+
+if not HAS_DOCKER_PY:
+    # No docker-py. Create a place holder client to allow
+    # instantiation of AnsibleModule and proper error handing
+    class Client(object):
+        def __init__(self, **kwargs):
+            pass
+
+
+def human_to_bytes(number):
```

Comment:
Maybe it should be, but i don't see anything right off in basic.py that turns a character string into an int. I see something called pretty_bytes() in basic.py which does the opposite of what I need. It turns a number into a string with this format: '%.2f %s' % (float(size)/ limit, suffix)